src/hotspot/share/opto/loopnode.cpp

```diff
@@ -5126,6 +5132,7 @@ int PhaseIdealLoop::build_loop_tree_impl( Node *n, int pre_order ) {
           _igvn.register_new_node_with_optimizer(halt);
           set_loop(halt, l);
           C->root()->add_req(halt);
+          _igvn._worklist.push(C->root());
```

Comment:
Maybe add a method to PhaseIterGVN that does the add_req + push similar to replace_input_of?

Reply:
@rwestrel Thanks for the suggestion, I will do that

Reply:
Yes, it is good suggestion.